vpr/src/device/rr_graph_view.h

```diff
@@ -62,6 +62,21 @@ class RRGraphView {
         return node_storage_.node_capacity(node);
     }
 
+    /* Get the direction of a routing resource node. This function is inlined for runtime optimization.
+     * INC_DIRECTION: wire driver is positioned at the low-coordinate end of the wire.
+     * DEC_DIRECTION: wire_driver is positioned at the high-coordinate end of the wire.
+     * BI_DIRECTION: wire has multiple drivers, so signals can travel either way along the wire
+     * INVALID_DIRECTION: wire does not have a valid direction set
+     */
+    inline e_direction node_direction(RRNodeId node) const {
+        return node_storage_.node_direction(node);
+    }
+
+    /* Get the direction string of a routing resource node. This function is inlined for runtime optimization. */
+    inline std::string node_direction_string(RRNodeId node) const {
+        return node_storage_.node_direction_string(node);
```

Comment:
Unless really really needed (e.g. if constructed on-the-fly), I'd try to avoid returning strings by value but rather return them by const reference.